### PR TITLE
LCAM-1401

### PR DESCRIPTION
**Pull request by Mhowell494**
[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-1401)

- Migrate validation scattered throughout the `manage_income_evidence` stored procedure into a centralised location within the evidence service.
- Unit test cases for the new validation checks.

### CODE REVIEW

File: crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/util/DateUtil.java
Comment (venkat980) [inline crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/util/DateUtil.java:1]: Maybe we can use the DateUtil from commons jar.

Comment (muthusubra) [reply crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/util/DateUtil.java:1]: Updated

File: crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/staticdata/enums/OtherEvidenceTypes.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    private final String evidence;
+
+    OtherEvidenceTypes(String evidence) {
+        this.evidence = evidence;
+    }
+
+    public String getEvidence() {
+        return evidence;
+    }
```
Comment (Mhowell494) [inline crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/staticdata/enums/OtherEvidenceTypes.java:20]: Can we use Lombok here to cut down on the boiler plate? :)

File: crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/service/IncomeEvidenceValidationService.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import java.time.ZoneId;
+import java.util.Date;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class IncomeEvidenceValidationService {
+
+    public void checkEvidenceReceivedDate(Date incomeEvidenceReceivedDate, Date applicationReceivedDate) {
+        Date currentDate = Date.from(LocalDateTime.now().atZone(ZoneId.systemDefault()).toInstant());
```
Comment (Mhowell494) [inline crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/service/IncomeEvidenceValidationService.java:20]: There's a method in the `crime-commons-classes` `DateUtil` that does this, probably better to reuse that.

**Review by Mhowell494 — COMMENTED** [review]
Looks good, just a few minor points.

Can we also please add a method in the Controller class e.g. `preProcessRequest()` that invokes these checks? I also think we could increase the code coverage a little.

File: crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/staticdata/enums/OtherEvidenceTypes.java (hunk 1 above)
Comment (muthusubra) [reply crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/staticdata/enums/OtherEvidenceTypes.java:20]: Updated

**Comment by muthusubra** [comment]
> preProcessRequest

As discussed offline, added a validate method in service class and added DTO with just the required fields

File: crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/service/IncomeEvidenceValidationService.java (hunk 2 above)
Comment (muthusubra) [reply crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/service/IncomeEvidenceValidationService.java]: Updated

**Comment by Mhowell494** [comment]
> > preProcessRequest
> 
> As discussed offline, added a validate method in service class and added DTO with just the required fields

**Review by Mhowell494 — COMMENTED** [review]
LG2M - can't approve as I created the PR

**Review by karlbaker02 — APPROVED** [review]
Approving on behalf of @Mhowell494

**Pull request merged**